diff --git a/backend/app/core/config.py b/backend/app/core/config.py
index 6a8ca50..650b9f7 100644
--- a/backend/app/core/config.py
+++ b/backend/app/core/config.py
@@ -75,7 +75,7 @@ class Settings(BaseSettings):
     SMTP_USER: str | None = None
     SMTP_PASSWORD: str | None = None
     EMAILS_FROM_EMAIL: EmailStr | None = None
-    EMAILS_FROM_NAME: EmailStr | None = None
+    EMAILS_FROM_NAME: str | None = None
 
     @model_validator(mode="after")
     def _set_default_emails_from(self) -> Self:
