diff --git a/src/app/page.tsx b/src/app/page.tsx
index 38eda1deac..24a27116d0 100644
--- a/src/app/page.tsx
+++ b/src/app/page.tsx
@@ -1,8 +1,9 @@
 'use client';
 import { redirect } from 'next/navigation';
+import { getItem } from '@/lib/storage';
 import { useEffect } from 'react';
 import { LAST_TEAM_CONFIG } from '@/lib/constants';
-import { getItem, removeItem } from '@/lib/storage';
+import { getItem } from '@/lib/storage';
 
 export default function RootPage() {
   useEffect(() => {
@@ -11,8 +12,6 @@ export default function RootPage() {
     if (lastTeam) {
       redirect(`/teams/${lastTeam}/websites`);
     } else {
-      removeItem(LAST_TEAM_CONFIG);
-
       redirect(`/websites`);
     }
   }, []);
diff --git a/src/components/input/NavButton.tsx b/src/components/input/NavButton.tsx
index 7ea966cc90..9552ac731e 100644
--- a/src/components/input/NavButton.tsx
+++ b/src/components/input/NavButton.tsx
@@ -93,7 +93,7 @@ export function NavButton({ showText = true }: TeamsButtonProps) {
                 <Column minWidth="300px">
                   <Menu selectionMode="single" selectedKeys={selectedKeys}>
                     <MenuSection title={formatMessage(labels.myAccount)}>
-                      <MenuItem id="user" href={getUrl('/')}>
+                      <MenuItem id="user" href={getUrl('/websites')}>
                         <IconLabel icon={<User />} label={user.username} />
                       </MenuItem>
                     </MenuSection>
