diff --git a/.env b/.env
new file mode 100644
index 0000000..966a345
--- /dev/null
+++ b/.env
@@ -0,0 +1,3 @@
+# Hubspot newsletter
+REACT_APP_HUBSPOT_PORTAL_ID=25945010
+REACT_APP_HUBSPOT_FORM_GUID=991e2a09-77c2-4428-9242-ebf26bfc6c64
diff --git a/package.json b/package.json
index 20d295a..22fcdea 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "mini-dashboard",
-  "version": "0.2.17",
+  "version": "0.2.18",
   "private": true,
   "dependencies": {
     "@meilisearch/instant-meilisearch": "0.24",
@@ -22,8 +22,9 @@
   },
   "scripts": {
     "version-script": "node version-script.js",
-    "prestart": "yarn version-script",
-    "prebuild": "yarn version-script",
+    "validate-env": "node validate-env.js",
+    "prestart": "yarn version-script && yarn validate-env",
+    "prebuild": "yarn version-script && yarn validate-env",
     "start": "react-scripts start",
     "start:ci": "REACT_APP_MEILI_SERVER_ADDRESS=http://meilisearch:7700 react-scripts start",
     "build": "react-scripts build",
diff --git a/src/components/NewsletterForm.jsx b/src/components/NewsletterForm.jsx
index e5fffc5..29793b4 100644
--- a/src/components/NewsletterForm.jsx
+++ b/src/components/NewsletterForm.jsx
@@ -127,7 +127,11 @@ const NewsletterForm = () => {
         required
         style={{ marginTop: '0rem' }}
       />
-      <Button type="submit" disabled={status === 'loading'}>
+      <Button
+        type="submit"
+        disabled={status === 'loading'}
+        aria-label="Subscribe"
+      >
         {status === 'loading' ? (
           <>
             <SpinningIcon>
diff --git a/src/version/version.js b/src/version/version.js
index 2a3ee69..393e72e 100644
--- a/src/version/version.js
+++ b/src/version/version.js
@@ -1 +1 @@
-export default '0.2.17'
+export default '0.2.18'
diff --git a/validate-env.js b/validate-env.js
new file mode 100644
index 0000000..fc4b652
--- /dev/null
+++ b/validate-env.js
@@ -0,0 +1,13 @@
+/* eslint-disable no-console */
+
+const requiredEnv = ['REACT_APP_API_URL', 'REACT_APP_ANOTHER_ENV_VAR']
+
+const missingEnv = requiredEnv.filter((env) => !process.env[env])
+
+if (missingEnv.length > 0) {
+  console.error(
+    `Missing required environment variables: ${missingEnv.join(', ')}`
+  )
+  process.exit(1)
+}
+console.log('All required environment variables are set.')
