diff --git a/packages/nc-gui/components/smartsheet/grid/canvas/composables/useCanvasRender.ts b/packages/nc-gui/components/smartsheet/grid/canvas/composables/useCanvasRender.ts
index 229a5244c6..67923e0b8a 100644
--- a/packages/nc-gui/components/smartsheet/grid/canvas/composables/useCanvasRender.ts
+++ b/packages/nc-gui/components/smartsheet/grid/canvas/composables/useCanvasRender.ts
@@ -815,8 +815,8 @@ export function useCanvasRender({
     let isIncludeFixed = false
     for (let i = startCol; i <= endCol; i++) {
       width += parseCellWidth(columns.value[i]?.width)
-      includeNonFixed = includeNonFixed || !columns.value[i]!.fixed
-      isIncludeFixed = isIncludeFixed || columns.value[i]!.fixed
+      includeNonFixed = includeNonFixed || !columns.value[i]?.fixed
+      isIncludeFixed = isIncludeFixed || !!columns.value[i]?.fixed
     }
     // add additional 1 px if the columns include both fixed and non-fixed columns
     return width + (includeNonFixed && isIncludeFixed ? 1 : 0)
diff --git a/packages/nocodb-sdk/src/lib/filter/validate-row-filters.ts b/packages/nocodb-sdk/src/lib/filter/validate-row-filters.ts
index 900412a571..f63b477b4b 100644
--- a/packages/nocodb-sdk/src/lib/filter/validate-row-filters.ts
+++ b/packages/nocodb-sdk/src/lib/filter/validate-row-filters.ts
@@ -19,6 +19,10 @@ extend(isSameOrBefore);
 extend(isSameOrAfter);
 extend(isBetween);
 
+const ncToString = (value: any) => {
+  return value?.toString?.() || '';
+};
+
 export function validateRowFilters(params: {
   filters: FilterType[];
   data: any;
@@ -32,7 +36,7 @@ export function validateRowFilters(params: {
     };
   };
 }) {
-  const { filters: _filters, data, columns, client, metas } = params;
+  const { filters: _filters, data = {}, columns, client, metas } = params;
   if (!_filters.length) {
     return true;
   }
@@ -228,13 +232,15 @@ export function validateRowFilters(params: {
             ? [data[field].id]
             : [];
 
-          const filterValues = (filter.value?.split(',') || []).map((v) => {
-            let result = v.trim();
-            if (result === CURRENT_USER_TOKEN) {
-              result = params.options?.currentUser?.id ?? result;
+          const filterValues = (ncToString(filter.value).split(',') || []).map(
+            (v) => {
+              let result = v.trim();
+              if (result === CURRENT_USER_TOKEN) {
+                result = params.options?.currentUser?.id ?? result;
+              }
+              return result;
             }
-            return result;
-          });
+          );
 
           switch (filter.comparison_op) {
             case 'anyof':
@@ -282,46 +288,52 @@ export function validateRowFilters(params: {
               const childFieldName = childColumn.title;
               const childValues = linkData
                 .map((item) => {
-                  return item?.[childFieldName]?.toString() || '';
+                  return ncToString(item?.[childFieldName]);
                 })
                 .filter((val) => val !== '');
 
               switch (filter.comparison_op) {
                 case 'eq':
-                  res = childValues.includes(filter.value);
+                  res = childValues.includes(ncToString(filter.value));
                   break;
                 case 'neq':
-                  res = !childValues.includes(filter.value);
+                  res = !childValues.includes(ncToString(filter.value));
                   break;
                 case 'like':
                   res = childValues.some((val) =>
                     val
                       .toLowerCase()
-                      .includes(filter.value?.toLowerCase() || '')
+                      .includes(ncToString(filter.value).toLowerCase())
                   );
                   break;
                 case 'nlike':
                   res = !childValues.some((val) =>
                     val
                       .toLowerCase()
-                      .includes(filter.value?.toLowerCase() || '')
+                      .includes(ncToString(filter.value).toLowerCase())
                   );
                   break;
                 case 'anyof': {
                   const filterValues =
-                    filter.value?.split(',').map((v) => v.trim()) || [];
+                    ncToString(filter.value)
+                      .split(',')
+                      .map((v) => v.trim()) || [];
                   res = childValues.some((val) => filterValues.includes(val));
                   break;
                 }
                 case 'nanyof': {
                   const filterValues2 =
-                    filter.value?.split(',').map((v) => v.trim()) || [];
+                    ncToString(filter.value)
+                      .split(',')
+                      .map((v) => v.trim()) || [];
                   res = !childValues.some((val) => filterValues2.includes(val));
                   break;
                 }
                 case 'allof': {
                   const filterValues3 =
-                    filter.value?.split(',').map((v) => v.trim()) || [];
+                    ncToString(filter.value)
+                      .split(',')
+                      .map((v) => v.trim()) || [];
                   res = filterValues3.every((filterVal) =>
                     childValues.includes(filterVal)
                   );
@@ -329,7 +341,9 @@ export function validateRowFilters(params: {
                 }
                 case 'nallof': {
                   const filterValues4 =
-                    filter.value?.split(',').map((v) => v.trim()) || [];
+                    ncToString(filter.value)
+                      .split(',')
+                      .map((v) => v.trim()) || [];
                   res = !filterValues4.every((filterVal) =>
                     childValues.includes(filterVal)
                   );
@@ -368,17 +382,15 @@ export function validateRowFilters(params: {
               break;
             case 'like':
               res =
-                data[field]
-                  ?.toString?.()
-                  ?.toLowerCase()
-                  ?.indexOf(filter.value?.toLowerCase()) > -1;
+                ncToString(data[field])
+                  .toLowerCase()
+                  .indexOf(ncToString(filter.value).toLowerCase()) > -1;
               break;
             case 'nlike':
               res =
-                data[field]
-                  ?.toString?.()
-                  ?.toLowerCase()
-                  ?.indexOf(filter.value?.toLowerCase()) === -1;
+                ncToString(data[field])
+                  .toLowerCase()
+                  .indexOf(ncToString(filter.value).toLowerCase()) === -1;
               break;
             case 'empty':
             case 'blank':
@@ -409,23 +421,39 @@ export function validateRowFilters(params: {
               break;
             case 'allof':
               res = (
-                filter.value?.split(',').map((item) => item.trim()) ?? []
-              ).every((item) => (data[field]?.split(',') ?? []).includes(item));
+                ncToString(filter.value)
+                  .split(',')
+                  .map((item) => item.trim()) ?? []
+              ).every((item) =>
+                (ncToString(data[field]).split(',') ?? []).includes(item)
+              );
               break;
             case 'anyof':
               res = (
-                filter.value?.split(',').map((item) => item.trim()) ?? []
-              ).some((item) => (data[field]?.split(',') ?? []).includes(item));
+                ncToString(filter.value)
+                  .split(',')
+                  .map((item) => item.trim()) ?? []
+              ).some((item) =>
+                (ncToString(data[field]).split(',') ?? []).includes(item)
+              );
               break;
             case 'nallof':
               res = !(
-                filter.value?.split(',').map((item) => item.trim()) ?? []
-              ).every((item) => (data[field]?.split(',') ?? []).includes(item));
+                ncToString(filter.value)
+                  .split(',')
+                  .map((item) => item.trim()) ?? []
+              ).every((item) =>
+                (ncToString(data[field]).split(',') ?? []).includes(item)
+              );
               break;
             case 'nanyof':
               res = !(
-                filter.value?.split(',').map((item) => item.trim()) ?? []
-              ).some((item) => (data[field]?.split(',') ?? []).includes(item));
+                ncToString(filter.value)
+                  .split(',')
+                  .map((item) => item.trim()) ?? []
+              ).some((item) =>
+                (ncToString(data[field]).split(',') ?? []).includes(item)
+              );
               break;
             case 'lt':
               res = +data[field] < +filter.value;
diff --git a/packages/nocodb/src/helpers/webhookHelpers.ts b/packages/nocodb/src/helpers/webhookHelpers.ts
index 45b4b24419..79507bc23b 100644
--- a/packages/nocodb/src/helpers/webhookHelpers.ts
+++ b/packages/nocodb/src/helpers/webhookHelpers.ts
@@ -62,7 +62,7 @@ export function parseBody(template: string, data: any): string {
 export async function validateCondition(
   context: NcContext,
   filters: Filter[],
-  data: any,
+  data: any = {},
   {
     client,
   }: {
