diff --git a/README.md b/README.md
index aa48bb8..5069a2d 100644
--- a/README.md
+++ b/README.md
@@ -114,7 +114,7 @@ Instant provides a [permissions layer](https://www.instantdb.com/docs/permission
   "invites": {
     "bind": [
       "isMember",
-      "auth.id in data.ref('team.memberships.userId')",
+      "auth.id in data.ref('teams.memberships.userId')",
       "isInvitee",
       "auth.email == data.userEmail"
     ],
@@ -126,7 +126,10 @@ Instant provides a [permissions layer](https://www.instantdb.com/docs/permission
     }
   },
   "drawings": {
-    "bind": ["isMember", "auth.id in data.ref('team.memberships.userId')"],
+    "bind": [
+      "isMember",
+      "auth.id in data.ref('teams.memberships.userId')"
+    ],
     "allow": {
       "view": "isMember",
       "create": "isMember",
@@ -137,9 +140,9 @@ Instant provides a [permissions layer](https://www.instantdb.com/docs/permission
   "memberships": {
     "bind": [
       "isMember",
-      "auth.id in data.ref('team.memberships.userId')",
+      "auth.id in data.ref('teams.memberships.userId')",
       "isInviteeOrCreator",
-      "size(data.ref('team.invites.id')) == 0 ? auth.id in data.ref('team.creatorId') : auth.email in data.ref('team.invites.userEmail')",
+      "size(data.ref('teams.invites.id')) == 0 ? auth.id in data.ref('teams.creatorId') : auth.email in data.ref('teams.invites.userEmail')",
       "isUser",
       "auth.id == data.userId"
     ],
diff --git a/src/mutators.ts b/src/mutators.ts
index 4353df1..b2c0923 100644
--- a/src/mutators.ts
+++ b/src/mutators.ts
@@ -13,7 +13,7 @@ export async function createDrawingForTeam({
 
   const result = await db.transact([
     tx.drawings[drawingId].merge({ name: drawingName ?? "Untitled" }),
-    tx.drawings[drawingId].link({ team: teamId }),
+    tx.drawings[drawingId].link({ teams: teamId }),
     tx.teams[teamId].link({ drawings: drawingId }),
   ]);
 
@@ -35,7 +35,7 @@ export async function createTeamWithMember({
   const result = await db.transact([
     tx.teams[teamId].update({ name: teamName, creatorId: userId }),
     tx.memberships[membershipId].update({ teamId, userId, userEmail }),
-    tx.memberships[membershipId].link({ team: teamId }),
+    tx.memberships[membershipId].link({ teams: teamId }),
   ]);
 
   return {
@@ -60,7 +60,7 @@ export async function inviteMemberToTeam({
 
   const result = await db.transact([
     tx.invites[inviteId].update({ userEmail, teamId, teamName }),
-    tx.invites[inviteId].link({ team: teamId }),
+    tx.invites[inviteId].link({ teams: teamId }),
   ]);
 
   return {
@@ -82,7 +82,7 @@ export async function acceptInvite({
 
   const result = await db.transact([
     tx.memberships[membershipId].update({ teamId, userId, userEmail }),
-    tx.memberships[membershipId].link({ team: teamId }),
+    tx.memberships[membershipId].link({ teams: teamId }),
   ]);
 
   return {
diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index 7bcc00d..c4baccc 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -195,7 +195,7 @@ function Team({
     drawings: {
       $: {
         where: {
-          "team.id": team.id,
+          "teams.id": team.id,
         },
         limit: drawingsPerPage,
         offset: drawingsPerPage * (pageNumber - 1),
@@ -204,7 +204,7 @@ function Team({
     memberships: {
       $: {
         where: {
-          "team.id": team.id,
+          "teams.id": team.id,
         },
       },
     },
@@ -222,7 +222,7 @@ function Team({
     drawings: {
       $: {
         where: {
-          "team.id": team.id,
+          "teams.id": team.id,
         },
         limit: drawingsPerPage + 1,
         offset: drawingsPerPage * (pageNumber - 1),
